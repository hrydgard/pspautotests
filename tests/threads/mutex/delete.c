#include "shared.h"

inline void deleteTest(const char *title, SceUID mutex) {
	int result = sceKernelDeleteMutex(mutex);
	if (result == 0) {
		printf("%s: OK\n", title);
	} else {
		printf("%s: Failed (%X)\n", title, result);
	}
}

int main(int argc, char **argv) {
	SceUID mutex = sceKernelCreateMutex("delete", 0, 0, NULL);

	deleteTest("Normal", mutex);
	deleteTest("NULL", 0);
	deleteTest("Invalid", 0xDEADBEEF);
	deleteTest("Deleted", mutex);

	BASIC_SCHED_TEST("Delete other",
		result = sceKernelDeleteMutex(mutex2);
	);
	BASIC_SCHED_TEST("Delete same",
		result = sceKernelDeleteMutex(mutex1);
	);
	BASIC_SCHED_TEST("NULL",
		result = sceKernelDeleteMutex(0);
	);

	return 0;
}