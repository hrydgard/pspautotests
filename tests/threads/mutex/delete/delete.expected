Normal: OK
NULL: Failed (800201C3)
Invalid: Failed (800201C3)
Deleted: Failed (800201C3)
Delete other: A1B1C2E1D1F2 (thread=800201B5, main=00000000)
Delete same: A1B1C2D1E2F1 (thread=800201B5, main=00000000)
NULL: A1B1C2E1D1F2 (thread=800201B5, main=800201C3)
